### ISSUE TITLE
False positive on "Variable assignment found within a conditional."

### ISSUE THREAD

**Issue by WraithKenny** [WordPress-Coding-Standards/WordPress-Coding-Standards#1337]
squizlabs/php_codesniffer: 3.2.3
wp-coding-standards/wpcs: 0.14.1
PHP 7.1.14 (cli)
```php
// No false-positive.
add_action( 'template_redirect', function() {
	$anything = false;
	$foobar   = ( $foobad ) ?: '';
} );

/**
 * No false-positive
 */
function test() {
	$anything = false;
	$foobar   = $foobad ?: '';
}

// Broken.
add_action( 'template_redirect', function() {
	$anything = false;
	$foobar   = $foobad ?: '';
} );
```

The bottom 2 assignments both show the error "Variable assignment found within a conditional."

**Comment by jrfnl** [comment]
Closing as a duplicate of #1227 which was fixed in #1249. The fix will be included in the next release.
